editor/filesystem_dock.cpp

```diff
@@ -182,16 +182,40 @@ bool FileSystemDock::_create_tree(TreeItem *p_parent, EditorFileSystemDirectory
 	// Create a tree item for the subdirectory.
 	TreeItem *subdirectory_item = tree->create_item(p_parent);
 	String dname = p_dir->get_name();
+	String lpath = p_dir->get_path();
+
 	if (dname.is_empty()) {
 		dname = "res://";
 	}
 
+	// Set custom folder color (if applicable).
+	Dictionary custom_folder_colors = ProjectSettings::get_singleton()->get_setting("file_customization/folder_colors");
```

Comment:
`custom_folder_colors` can be a member of FileSystemDock instead. It would allow to use `get_setting()` once, in the constructor, instead of for every folder.